Core roadmap June 2026 (#13079)
dbt-labs/dbt-core: docs/roadmap/2026-06-announcing-v2.md

# Announcing… v2.0 (June 2026)

*At last.*

> Okay, so you saw the [announcement](https://www.getdbt.com/blog/fivetran-and-dbt-are-one-company-now-here-s-what-that-means), and you read [the dev blog post](https://docs.getdbt.com/blog/dbt-core-v2-is-here), which explain in detail how we got from there (Fusion, new engine, ELv2) to here (it’s just dbt Core, v2.0).

Reviewer: ```suggestion
Okay, so you saw the [announcement](https://www.getdbt.com/blog/fivetran-and-dbt-are-one-company-now-here-s-what-that-means), and you read [the dev blog post](https://docs.getdbt.com/blog/dbt-core-v2-is-here), which explains in detail how we got from there (Fusion, new engine, ELv2) to here (it’s just dbt Core, v2.0).
```

It was only through this hard work that we proved the “core” of the Rust rewrite could be — *should* be — dbt Core v2.0. That it no longer made sense to keep living in a bifurcated world, when we can make a much better one for everyone to live in.

Still… we must have had an inkling, right? Heck, it’s been `dbt-fusion==2.0.0.xxx` since the very first release, in May 2025. (I guess the we had the date right, just the year was [OB1](https://en.wikipedia.org/wiki/Off-by-one_error).) Today, the foundational code is available as dbt Core v2.0. *One engine, under the Apache 2.0 license, indivisible, with faster parsing and modern interfaces for all.* 

In the meantime, we actually did the two-engine thing. We triaged bugs and feature requests. We refined them, shipped them, and tested them, *twice,* in both dbt Core v1.12 (Python) and Fusion / v2.0 (Rust). So, regardless of whether you’re still using Core or Fusion today, we’ve got good news for you - the dbt framework is alive and well.

| **Version** | **When** | **Namesake** | **Stuff** |
| --- | --- | --- | --- |

> | [v1.12](https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.10) | May (beta) | Alecia Beth Moore-Hart (P!NK) | UDF extensions. Easier specs for semantic layer, and Iceberg `catalogs`. Long-upvoted feature requests: `vars.yml`; `on_error` model config; [mix-and-match yaml selectors](https://docs.getdbt.com/reference/node-selection/methods?version=1.12#selector); `run-operation` for adhoc `—-sql`; jinja file extensions; `latest_version_pointer`. Improved exception handling and error messages. Dozens of bug fixes across core + adapters, including Python 3.14 support. `dbt login` to unlock paid platform features. `--use-v2-parser` to try out the speedy new Rust parser before you upgrade to v2.0. |

Author: ```suggestion
| [v1.12](https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.12) | May (beta) | Alecia Beth Moore-Hart (P!NK) | UDF extensions. Easier specs for semantic layer, and Iceberg `catalogs`. Long-upvoted feature requests: `vars.yml`; `on_error` model config; [mix-and-match yaml selectors](https://docs.getdbt.com/reference/node-selection/methods?version=1.12#selector); `run-operation` for adhoc `—-sql`; jinja file extensions; `latest_version_pointer`. Improved exception handling and error messages. Dozens of bug fixes across core + adapters, including Python 3.14 support. `dbt login` to unlock paid platform features. `--use-v2-parser` to try out the speedy new Rust parser before you upgrade to v2.0. |
```

> | [v2.0](https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.11) | June (alpha) |  | Faster Rust rewrite, at full parity with v1. Strict codified language spec. New parquet metadata artifacts, built for speed & scale, powering new dbt-docs. |

Author: ```suggestion
| v2.0 | June (alpha) |  | Faster Rust rewrite, at full parity with v1. Strict codified language spec. New parquet metadata artifacts, built for speed & scale, powering new dbt-docs. |
```

Our team has been living this two-engine world first-hand: making the same bug fixes and feature implementations across two codebases, in two different languages and testing frameworks. That duplication of effort has definitely slowed us down, and risks divergence of behavior.

That's why we’re excited to move toward a world where one engine powers both the Core and Fusion distributions. Rather than needing to think and talk about a unified “language” distinct from an underlying “engine” that implements it — our beloved (now, dearly departed) donut diagram — it’s one ****foundation to power it all**.**

## Present

Today, we [announced the first alpha release of dbt Core v2.0](https://docs.getdbt.com/blog/dbt-core-v2-is-here). This new major version is powered by the same Rust rewrite of Core v1 that powers Fusion. All of the code is licensed under Apache 2.0, and you can see it on the `main` branch of the `dbt-core` repo. 

This is the future of the dbt framework:

- fast and built for scale (Rust, ADBC, etc)
- shared foundation - one donut, not two donuts
- single adapter layer - all bundled in

> - and one more thing… a brand-new dbt-docs (!) This is a long-awaited refresh (something we talked about in our [very first roadmap post](https://github.com/dbt-labs/dbt-core/blob/main/docs/roadmap/2022-05-dbt-a-core-story.md#epilogue-whats-missing)) - check out the github discussion (link coming soon) for more details

Author: ```suggestion
- and one more thing… a brand-new dbt-docs (!) This is a long-awaited refresh (something we talked about in our [very first roadmap post](https://github.com/dbt-labs/dbt-core/blob/main/docs/roadmap/2022-05-dbt-a-core-story.md#epilogue-whats-missing)) - check out [the github discussion](https://github.com/dbt-labs/dbt-core/discussions/13080) for more details
```

Reviewer: https://github.com/dbt-labs/dbt-core/discussions/13080

> - and one more thing… a brand-new dbt-docs (!) This is a long-awaited refresh (something we talked about in our [very first roadmap post](https://github.com/dbt-labs/dbt-core/blob/main/docs/roadmap/2022-05-dbt-a-core-story.md#epilogue-whats-missing)) - check out the github discussion (link coming soon) for more details

Reviewer: ```suggestion
- and one more thing… a brand-new dbt-docs (!) This is a long-awaited refresh (something we talked about in our [very first roadmap post](https://github.com/dbt-labs/dbt-core/blob/main/docs/roadmap/2022-05-dbt-a-core-story.md#epilogue-whats-missing)) - check out [the github discussion](https://github.com/dbt-labs/dbt-core/discussions/13080) for more details
```

If you want to read more about how our thinking and licensing strategy evolved over the past year, [Joel & Grace wrote a whole blog post](https://docs.getdbt.com/blog/dbt-core-v2-is-here).

**Why alpha?** Isn’t Fusion already in Preview? 

We’ve spent the last 6+ months working hard on Fusion’s “conformance”: given the same project inputs, does it produce the same results as Core v1.X? We fixed thousands of bugs using this method, and while we now feel confident running Fusion in an environment we manage (dbt platform), software loves to find new edge cases out in the big bold world. 

That's why we need YOUR help to run and test v2.0 against the full population of dbt projects. We need YOUR ever-bigger DAGs, YOUR wacky package macros, YOUR bespoke override of the incremental materialization, and YOUR 2009 laptop running Windows Vista.

> The best way to help is by [testing out v2.0](https://docs.getdbt.com/docs/local/install-dbt) while it’s in alpha - try running it against your project, and let us know if you encounter any unexpected errors

Author: ```suggestion
The best way to help is by [testing out v2.0](https://docs.getdbt.com/docs/local/install-dbt) while it’s in alpha - try running it against your project, and let us know if you encounter any unexpected errors.
```

- If you’re an **adapter maintainer:** don’t you worry, check out [this Guide](https://docs.getdbt.com/guides/adapter-creation-v2) about how to contribute support for your favorite DWH to the newly verticalized adapter layer in Core v2.0.

### What’s next for the dbt framework?

We’re going to keep building the dbt framework — new capabilities and improvements of existing ones, on top of the v2.0 engine. As always, we'll be monitoring github for your comments and upvotes, but there are a handful of feature ideas already on our minds:

1. **Model freshness:** Check to see if your models are as fresh as you think they should be, within *and across* projects, so that data product consumers can check if producers are meeting a promised SLA. (And before you ask - yes, we did talk about this in our last roadmap post as a thing we were planning to include in v1.12; we had to make some tricky trade-offs as part of doing the v2.0 effort, and this was one of them. But we’re still interested! If you are too, [give the issue a read](https://github.com/dbt-labs/dbt-core/issues/12719) and let us know what you think.)

> 2. **Checks:** In the dbt-docs discussion (link coming soon), we describe how we’re powering fast local metadata queries with a new set of well-structured parquet files and an in-process database. Those same files could power *project quality checks* as metadata queries, folded into the task graph, executed during compile / before execution. You could prevent a `dbt build` if a model is missing an owner, messing up a naming convention, or selecting from a source it shouldn’t be. (For any SDF knowers out there, [this should sound familiar](https://docs.sdf.com/guide/data-quality/checks).) Imagine: a [v2.0 of the dbt-project-evaluator package](https://github.com/dbt-labs/dbt-project-evaluator) that ships our best practices as `checks`, written as simple SQL queries (instead of Jinja `{{ graph }}` manipulation that’s as impressive as it is illegible).

Author: ```suggestion
2. **Checks:** In [the dbt-docs discussion](https://github.com/dbt-labs/dbt-core/discussions/13080), we describe how we’re powering fast local metadata queries with a new set of well-structured parquet files and an in-process database. Those same files could power *project quality checks* as metadata queries, folded into the task graph, executed during compile / before execution. You could prevent a `dbt build` if a model is missing an owner, messing up a naming convention, or selecting from a source it shouldn’t be. (For any SDF knowers out there, [this should sound familiar](https://docs.sdf.com/guide/data-quality/checks).) Imagine: a [v2.0 of the dbt-project-evaluator package](https://github.com/dbt-labs/dbt-project-evaluator) that ships our best practices as `checks`, written as simple SQL queries (instead of Jinja `{{ graph }}` manipulation that’s as impressive as it is illegible).
```

We want to acknowledge that *this is a change.* It’s a change that feels good to us, because it’s a way to give the community easier access to all parts of dbt, and to maintain *fewer better channels* for product feedback. But that change might feel weird for some of you. If that’s you, we want to talk about it - DM us in dbt community slack (really).

## Mergin’

Today is a big day for the dbt community:

- We’ve merged two companies (Fivetran + dbt Labs) into one (name TBD!)
- We’re merging two engines (Fusion + Core) into one (Core v2.0)

We are once again committing to true-blue Apache-2 OSS as the foundation, and the future, of dbt.

Our request for all of you:

- be excited with us and talk about it
- try this all out and tell us how to make it better
- as we consolidate repos, help us make sure no long-loved issue is forgotten
- shape the future of dbt by joining [discussions on GitHub](https://github.com/dbt-labs/dbt-core/discussions) 
- and, hey, [come hangout in Vegas this September](https://www.getdbt.com/dbt-summit/registration)

> HAGS,

Author: ```suggestion
HAGS (have a great summer),
```
